chain/client/tests/process_blocks.rs

```diff
@@ -2745,6 +2761,45 @@ fn test_congestion_receipt_execution() {
     }
 }
 
+#[cfg(feature = "protocol_feature_cap_max_gas_price")]
+#[cfg(test)]
+mod cap_max_gas_price_tests {
+    use super::*;
+
+    fn does_gas_price_exceed_limit(protocol_version: ProtocolVersion) -> bool {
+        let mut env =
+            create_env_with_congestion(protocol_version, Some(Rational::new_raw(2, 1)), 7).0;
+
+        for i in 3..20 {
+            env.produce_block(0, i);
+            let block = env.clients[0].chain.get_block_by_height(i).unwrap().clone();
+            let protocol_version = env.clients[0]
+                .runtime_adapter
+                .get_epoch_protocol_version(block.header().epoch_id())
+                .unwrap();
+            let min_gas_price =
+                env.clients[0].chain.block_economics_config.min_gas_price(protocol_version);
+            if block.header().gas_price() > 10 * min_gas_price {
+                return true;
```

Comment:
Please also assert that congestion actually happens. You can do this by asserting that gas usage is at least as large as gas limit.